### PR TITLE
Colt Single Action Army

### PR DESCRIPTION
**Pull request by Shadowmech88**
coltgrey coltgreycocked

Adds the Colt Single Action Army, a single-action .357 revolver that must be manually cocked via `attack_self()` before each shot. Cartridges can be removed via Alt+click.
The revolver can be spun if you `attack_self()` while it's cocked.

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 1]
```diff
@@ -233,3 +233,34 @@
...
+	else
+		icon_state = "colt"
+
+/obj/item/weapon/gun/projectile/colt/attack_self(mob/user, var/callparent = FALSE)
+	if(callparent)
+		return ..(user)
+	if(!cocked)
+		cocked = TRUE
+		update_icon()
+		to_chat(user, "You cock \the [src].")
```
Comment (Iamgoofball) [inline code/modules/projectiles/guns/projectile/revolver.dm:265]: no spinning 0/10

Comment (Shadowmech88) [reply code/modules/projectiles/guns/projectile/revolver.dm:265]: I ran out of ways to interact with the revolver.
I feel like using a verb would be less than optimal.

Comment (Iamgoofball) [reply code/modules/projectiles/guns/projectile/revolver.dm:265]: make it spin if it's cocked and you click on it in-hand

Comment (Iamgoofball) [reply code/modules/projectiles/guns/projectile/revolver.dm:265]: also moving it to/from inventory slots

Comment (Iamgoofball) [reply code/modules/projectiles/guns/projectile/revolver.dm:265]: dont forget to spin the inhand icon too

**Review by ComicIronic — APPROVED** [review]

**Comment by SonixApache** [comment]
Does it ricochet?

**Comment by Militaires** [comment]
you will now know why they call me "revolver"

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 2]
```diff
@@ -233,3 +233,63 @@
...
+	var/last_spin = 0
+	var/spin_delay = 1 SECONDS	//let's not get crazy
+
+/obj/item/weapon/gun/projectile/colt/update_icon()
+	if(cocked)
+		icon_state = "colt_cocked"
+	else
+		icon_state = "colt"
+
+/mob/living/carbon/update_transform()
```
Comment (ComicIronic) [inline code/modules/projectiles/guns/projectile/revolver.dm]: What is this code doing here?

Comment (Shadowmech88) [reply code/modules/projectiles/guns/projectile/revolver.dm]: My bad, I copied it as a reference for working with transforms.

**Comment by Shadowmech88** [comment]
Fixed.

**Comment by Probe1** [comment]
It's a "Put DNM on a PR then disappear for a week" episode.

**Pull request merged**